### PR TITLE
removed mode='auto' option

### PR DESCRIPTION
**Pull request by naylor-b**

### CODE REVIEW

File: openmdao/core/driver.py [hunk 1]
```diff
@@ -173,14 +174,23 @@ def _setup_driver(self, problem):
...
+            response_size += data['size']
 
         # Gather up the information for design vars.
         self._designvars = model.get_design_vars(recurse=True)
+        desvar_size = np.sum(data['size'] for data in itervalues(self._designvars))
+
+        if ((problem._mode == 'fwd' and desvar_size > response_size) or
+                (problem._mode == 'rev' and response_size > desvar_size)):
+            warnings.warn("Inefficient choice of derivative mode.  You chose '%s' for a "
+                          "design variable size of %d and a response size of %d." %
```
Comment (hwangjt) [inline openmdao/core/driver.py]: For clarity, I would recommend rephrasing 'response size' and 'design variable size'. A suggestion is: 'You chose ... for a problem with %d design variables and %d response variables (e.g., objective and constraints).'

File: openmdao/docs/feature_reference/core_features/working_with_derivatives/picking_mode.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+Pass 'fwd' to indicate forward mode and 'rev' to indicate reverse mode.
+
+If you choose a mode that is not optimal based on the sizes in your problem, you
+will see a warning like this:
+
+
+.. code-block:: none
+
+    RuntimeWarning: Inefficient choice of derivative mode.  You chose 'rev' for a
+    design variable size of 100 and a response size of 200.
```
Comment (hwangjt) [inline openmdao/docs/feature_reference/core_features/working_with_derivatives/picking_mode.rst]: My other comment applies here as well

File: openmdao/core/driver.py (hunk 1 above)
Comment (naylor-b) [reply openmdao/core/driver.py]: done

File: openmdao/docs/feature_reference/core_features/working_with_derivatives/picking_mode.rst (hunk 2 above)
Comment (naylor-b) [reply openmdao/docs/feature_reference/core_features/working_with_derivatives/picking_mode.rst]: done

**Pull request merged**